site/topics/numbers/twos-complement.rst

```diff
@@ -245,10 +245,247 @@ Problems and Limitations
+      - ``0``
+      - ``1``
+      - ``0``
+      - ``1``
+      -
+      - :math:`5`
+    * - **Number 2**
+      -
+      - ``1``
+      - ``0``
+      - ``1``
+      - ``0``
+      -
+      - :math:`-5`
+    * - **Sum**
+      -
+      - ``1``
+      - ``1``
+      - ``1``
+      - ``1``
+      -
+      - :math:`-0`
+
+
+* The above example results in a correct addition, but issues arise with non zero results
+
+.. note::
+
+    With binary addition and a fixed number of bits, any carry that results in an overflow will ultimately be ignored.
+    For example, consider the four bit addition of ``1111`` (:math:`7`) + ``0001`` (:math:`1`). The result is clearly
+    ``10000`` (:math:`8`), but since there are only four bits available, the overflowed value is lost, and the result
+    would be ``0000``.
+
+
+.. list-table:: Adding a negative binary number with ones complement
+    :widths: auto
+    :align: center
+
+    * - **Carry**
+      -
+      - ``1``
+      -
+      -
+      -
+      -
+      -
+    * - **Number 1**
+      -
+      - ``0``
+      - ``1``
+      - ``0``
+      - ``1``
+      -
+      - :math:`5`
+    * - **Number 2**
+      -
+      - ``1``
+      - ``1``
+      - ``0``
+      - ``0``
+      -
+      - :math:`-3`
+    * - **Sum**
+      -  (ignore overflow) ``1``
+      - ``0``
+      - ``0``
+      - ``0``
+      - ``1``
+      -
+      - :math:`1?`
+
+
+.. list-table:: Adding a negative binary number with ones complement
+    :widths: auto
+    :align: center
+
+    * - **Carry**
+      -
+      - ``1``
+      -
+      -
+      -
+      -
+      -
+    * - **Number 1**
+      -
+      - ``1``
+      - ``1``
+      - ``0``
+      - ``1``
+      -
+      - :math:`-2`
+    * - **Number 2**
+      -
+      - ``0``
+      - ``1``
+      - ``1``
+      - ``0``
+      -
+      - :math:`6`
+    * - **Sum**
+      -  (ignore overflow) ``1``
+      - ``0``
+      - ``0``
+      - ``1``
+      - ``1``
+      -
+      - :math:`3?`
+
+
+.. list-table:: Adding a negative binary number with ones complement
+    :widths: auto
+    :align: center
+
+    * - **Carry**
+      -
+      - ``1``
+      - ``1``
+      - ``1``
+      -
+      -
+      -
+    * - **Number 1**
+      -
+      - ``1``
+      - ``0``
+      - ``1``
+      - ``1``
+      -
+      - :math:`-4`
+    * - **Number 2**
+      -
+      - ``1``
+      - ``1``
+      - ``0``
+      - ``1``
+      -
+      - :math:`-2`
+    * - **Sum**
+      -  (ignore overflow) ``1``
+      - ``1``
+      - ``0``
+      - ``0``
+      - ``0``
+      -
+      - :math:`-7?`
+
+
+* The fact that the overflow value is lost is important for addition of negative numbers to work
+* However, when looking at these results, it is clear that addition with negative numbers doesn't quite work
+* Except for when the sum is :math:`0`, all results are one less than what they should be
+* In fact, considering :math:`-0` is a strange number, maybe the sum should be :math:`0`, which is also off by one
```

Comment:
I don't think I ever noticed that. My brain was just like "two 2s complement is cool"